fdbcli/DataDistributionCommand.actor.cpp

```diff
@@ -61,13 +61,28 @@ ACTOR Future<Void> setDDMode(Reference<IDatabase> db, int mode) {
 	}
 }
 
-ACTOR Future<Void> setDDIgnoreRebalanceSwitch(Reference<IDatabase> db, bool ignoreRebalance) {
+ACTOR Future<Void> setDDIgnoreRebalanceSwitch(Reference<IDatabase> db, uint8_t DDIgnoreOptionMask, bool setMaskedBit) {
 	state Reference<ITransaction> tr = db->createTransaction();
 	loop {
 		tr->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
+		tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
 		try {
-			if (ignoreRebalance) {
-				tr->set(fdb_cli::ddIgnoreRebalanceSpecialKey, ValueRef());
+			state ThreadFuture<Optional<Value>> resultFuture = tr->get(rebalanceDDIgnoreKey);
+			Optional<Value> v = wait(safeThreadFutureToFuture(resultFuture));
+			uint8_t oldValue = 0; // nothing is disabled
+			if (v.present()) {
+				if (v.get().size() > 0) {
+					oldValue = BinaryReader::fromStringRef<uint8_t>(v.get(), Unversioned());
+				} else {
+					// In old version (<= 7.1), the value is an empty string, which means all DD rebalance functions are
+					// disabled
+					oldValue = DDIgnore::ALL;
+				}
+				// printf("oldValue: %d Mask: %d V:%d\n", oldValue, DDIgnoreOptionMask, v.get().size());
+			}
+			uint8_t newValue = setMaskedBit ? (oldValue | DDIgnoreOptionMask) : (oldValue & ~DDIgnoreOptionMask);
+			if (newValue > 0) {
+				tr->set(fdb_cli::ddIgnoreRebalanceSpecialKey, BinaryWriter::toValue(newValue, Unversioned()));
```

Comment:
What happens after downgrade?

Reply:
I think in the old version, it doesn't matter what the value is, the old version FDB version just cares about whether this key exists.